src/main/java/io/cryostat/discovery/ContainerDiscovery.java

```diff
@@ -291,73 +354,67 @@ private CompletableFuture<ContainerDetails> doContainerInspectRequest(ContainerS
         return result;
     }
 
+    private void handleContainerEvent(List<ContainerSpec> current) {
+        List<DiscoveryNode> targetNodes =
+                DiscoveryNode.findAllByNodeType(BaseNodeType.JVM).stream()
+                        .filter(
+                                (n) ->
+                                        Objects.nonNull(n.target)
+                                                && getRealm()
+                                                        .equals(
+                                                                n.target
+                                                                        .annotations
+                                                                        .cryostat()
+                                                                        .get("REALM")))
+                        .collect(Collectors.toList());
+
+        Map<Target, ContainerSpec> containerRefMap = new HashMap<>();
+        current.stream()
+                .map((desc) -> Pair.of(desc, toTarget(desc)))
+                .filter((pair) -> Objects.nonNull(pair.getRight()))
+                .forEach((pair) -> containerRefMap.put(pair.getRight(), pair.getLeft()));
+
+        Set<Target> persistedTargets =
+                targetNodes.stream().map((n) -> n.target).collect(Collectors.toSet());
+        Set<Target> observedTargets = containerRefMap.keySet();
+
+        Target.compare(persistedTargets)
+                .to(observedTargets)
+                .added()
+                .forEach(
+                        (t) ->
+                                notify(
+                                        ContainerDiscoveryEvent.from(
+                                                containerRefMap.get(t), t, EventKind.FOUND)));
+
+        Target.compare(persistedTargets)
+                .to(observedTargets)
+                .removed()
+                .forEach(
+                        (t) ->
+                                notify(
+                                        ContainerDiscoveryEvent.from(
+                                                containerRefMap.get(t), t, EventKind.LOST)));
+    }
+
+    private void notify(ContainerDiscoveryEvent evt) {
+        bus.publish(CONTAINER_DISCOVERY_ADDRESS, evt);
+    }
+
     @Transactional
-    public void handleContainerEvent(ContainerSpec desc, EventKind evtKind) {
+    @ConsumeEvent(value = "io.cryostat.discovery.ContainerDiscovery", blocking = true)
```

Comment:
You can do this:

```
class A {
  notify() {
    bus.publish(B.class.getName(), msg);
  }
}

class B {
  @ConsumeEvent
  void onMessage(T msg) {
  }
}
```

ie. if you don't specify `value = "pkg.Class"` in the `@ConsumeEvent`, it defaults to the class name. Then you can just publish to the class name.

Reply:
Oh yes thanks! I was just investigating why the message is not being received...